connectors/es/index.py

```diff
@@ -40,7 +40,32 @@ def _create_object(self, doc):
         """
         raise NotImplementedError
 
-    async def get_all_docs(self, query=None, page_size=DEFAULT_PAGE_SIZE):
+    async def fetch_by_id(self, doc_id):
+        await self.client.indices.refresh(index=self.index_name)
```

Comment:
I'm curious about it - should we always call it or should we never call it?

I _think_ we do refresh stuff automatically, but also awaiting a `refresh` does not guarantee that the data is refreshed, only that the `refresh` operation was issued and accepted, is it correct?

Reply:
The `refresh` call is synchronous, so it's guaranteed that we get refreshed data.

For most of the use cases (if not all), we use `search`. When we use `fetch_by_id`, it normally means we need to refresh the document, because there will likely be an update happened. That's why we need the refresh.

Yes, ES does auto-refresh every second by default, and the official guide suggests not to use the refresh API, but [use `refresh=wait_for` when updating](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html#refresh-api-desc).

[`refresh=wait_for`](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html) will wait for a periodic refresh to happen before returning. If we apply this option to our update request, it may take, the worst case 1 second, to get the response. cc @tarekziade